src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1397,22 +1543,40 @@ codet java_bytecode_convert_methodt::convert_instructions(
       if(!i_it->source_location.get_line().empty())
         java_new_array.add_source_location()=i_it->source_location;
 
+      c=code_blockt();
+      if(!disable_runtime_checks)
+      {
+        // TODO make this throw NegativeArrayIndexException instead.
+        constant_exprt intzero=as_number(0, java_int_type());
```

Comment:
probably a relic from some-time, if I understand correctly, than both fill `java_int_type` with leading 0, so there should be no difference, therefore I changed it